components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -596,43 +585,43 @@ abstract class AbstractFetchDownloadService : Service() {
      * Updates the status of an ACTIVE download to completed or failed based on bytes copied
      */
     internal fun verifyDownload(download: DownloadJobState) {
-        if (getDownloadJobStatus(download) == DownloadJobStatus.ACTIVE &&
-            download.currentBytesCopied < download.state.contentLength ?: 0) {
-            setDownloadJobStatus(download, FAILED)
+        if (getDownloadJobStatus(download) == Status.ACTIVE &&
+            download.state.currentBytesCopied < download.state.contentLength ?: 0) {
+            setDownloadJobStatus(download, Status.FAILED)
             logger.error("verifyDownload for ${download.state.url} FAILED")
-        } else if (getDownloadJobStatus(download) == DownloadJobStatus.ACTIVE) {
-            setDownloadJobStatus(download, COMPLETED)
+        } else if (getDownloadJobStatus(download) == Status.ACTIVE) {
+            setDownloadJobStatus(download, Status.COMPLETED)
             /**
              * In cases when we don't get the file size provided initially, we have to
              * use downloadState.currentBytesCopied as a fallback.
              */
             val fileSizeNotFound = download.state.contentLength == null || download.state.contentLength == 0L
             if (fileSizeNotFound) {
-                val newState = download.state.copy(contentLength = download.currentBytesCopied)
+                val newState = download.state.copy(contentLength = download.state.currentBytesCopied)
                 updateDownloadState(newState)
             }
-            logger.debug("verifyDownload for ${download.state.url} ${download.status}")
+            logger.debug("verifyDownload for ${download.state.url} ${download.state.status}")
         }
     }
 
     private fun copyInChunks(downloadJobState: DownloadJobState, inStream: InputStream, outStream: OutputStream) {
         val data = ByteArray(CHUNK_SIZE)
         logger.debug("starting copyInChunks ${downloadJobState.state.url}" +
-                " currentBytesCopied ${downloadJobState.currentBytesCopied}")
+                " currentBytesCopied ${downloadJobState.state.currentBytesCopied}")
 
         // To ensure that we copy all files (even ones that don't have fileSize, we must NOT check < fileSize
-        while (getDownloadJobStatus(downloadJobState) == DownloadJobStatus.ACTIVE) {
+        while (getDownloadJobStatus(downloadJobState) == Status.ACTIVE) {
             val bytesRead = inStream.read(data)
 
             // If bytesRead is -1, there's no data left to read from the stream
             if (bytesRead == -1) { break }
 
-            downloadJobState.currentBytesCopied += bytesRead
```

Comment:
You also have to update the store here, as we want this changes to be reflected there too, you could use  [updateDownloadState](https://github.com/mozilla-mobile/android-components/blob/cdef87e847f4272fcb43e441794013e37feb97b9/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L677) this [here you can see an example](https://github.com/mozilla-mobile/android-components/blob/cdef87e847f4272fcb43e441794013e37feb97b9/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt#L612)